tests|plotting: Some refactoring/improvements around `BlockTools`

xdustinface opened the pull request:
See commit messages.

xdustinface committed:
tests: Use `time_out_assert` to wait for plot refreshing

xdustinface committed:
tests: Make `plot_manager` member of `BlockTools`

This also:

- Moves some code 
related to refreshing into `BlockTools.refresh_plots`.
- Moves setting the public keys of `PlotManager` into `setup_keys.

xdustinface committed:
tests: Directly use the plot info, don't create a sorted list upfront

xdustinface committed:
tests: Refactor a loop to use `values` instead of `items`

xdustinface committed:
plotting: Return paths of created and existed plots in `create_plots`

xdustinface committed:
tests: Introduce `new_plot` in `BlockTools`

xdustinface committed:
tests: Use `BlockTools.new_plot` in `TestPoolWalletRpc`

xdustinface committed:
tests: Allow and fix `flake8` validation in `test_pool_rpc.py`

xdustinface committed:
tests: Introduce `BlockTools.delete_plot` and use it in pool tests

xdustinface committed:
tests: Cleanup plots potentially left from a failed run in pool tests

arvidn commented on tests/pools/test_pool_rpc.py:
```diff
@@ -775,7 +744,6 @@ async def status_is_self_pooling():
                 return pw_status.current.state == PoolSingletonState.SELF_POOLING.value
 
             await time_out_assert(timeout=WAIT_SECS, function=status_is_self_pooling)
-            pw_info: PoolWalletInfo = (await client.pw_status(wallet_id))[0]
```
I get the impression `flake8` complained about the variable not being used. Do you think it would make sense to still make the function call and discard the return value?

mariano54 replied:
This is just an artifact of a lot of debugging when these tests were not working (during pool development). These calls are not needed and can be safely removed.

arvidn reviewed (commented):
I think everything looks reasonable except for the last commit, that deletes plots.

I noticed that if creating the plots fail, we only clean them up if it was because of a KeyboardInterrupt. It would probably make sense to clean up *regarldless* of why it failed. Perhaps the last commit wouldn't be necessary anymore

mariano54 commented on tests/pools/test_pool_rpc.py:
```diff
@@ -51,6 +62,7 @@ async def two_wallet_nodes(self):
 
     @pytest.fixture(scope="function")
     async def one_wallet_node_and_rpc(self):
+        delete_plots()
```
We only deleted plots in a specific directory, that are used for this test. This test needs to create and use a new plot dynamically, so we want to make sure we clean up.

mariano54 reviewed (dismissed):
I like the changes to block tools

xdustinface commented:
> Previously we just deleted plots specific plots, never all of them, why is that necessary?

Yeah that's correct and that's what we also still do with `BlockTools.delete_plot`. Now with this PR i moved the plots specifically plotted for this test into a separate directory and the cleanup is to make sure there is no leftover from wherever because extra plots in there now lead to test failure in `test_callback` of `BlockTools`. I changed the last commit to just drop the whole directory on test start instead of just deleting all files in it, see last force push.

arvidn commented:
> Yeah that's correct and that's what we also still do with BlockTools.delete_plot. Now with this PR i moved the plots specifically plotted for this test into a separate directory and the cleanup is to make sure there is no leftover from wherever because extra plots in there now lead to test failure in test_callback of BlockTools

I see. I get it now!
